extension/background/browserAction.js

```diff
@@ -11,9 +11,13 @@ browserAction.show = function(callback, tab) {
 	}
 
 	if(data.popup) {
-		chrome.browserAction.setPopup({
-			tabId: tab.id,
-			popup: "popups/" + data.popup
+		chrome.tabs.get(tab.id, function() {
```

Comment:
Yes! Sometimes, tabId changes, causing a javascript error there. the chrome.tabs.get is a method to check the tabId exists.